rfd/0196-opinionated-intro.md

```diff
@@ -0,0 +1,148 @@
+  * I want to know why I would use a given feature or interface of the product, its strengths and weaknesses, to help form my own opinion about how to use it best in my organization.
+* As a Teleporter
+  * I want this intro to be easy to maintain and keep current with the product and docs.
+  * I want to measure usage of the intro.
+  * I want new admins to be exposed to the various interfaces of the product like command-line, web, Connect.
+
+### Rabbit Holes
+
+This implementation is limited to the persona of a new admin. In the future we should have similar intros for other personas like dev users, SSO users (an access use case similar to our own internal platform.teleport.sh), admins coming into an existing cluster, etc. Product will do future work with design and other teams to formulate these personas. This means we need to define “a new admin.” Since we do not have user types in the app, simply roles, we’ll need to define this via one or a combination of roles, and this should only show up for newly created clusters.
+
+### Out of Bounds
+
+## Outlines/Sketches
+
+There is an intro page that is the landing page for a new user, and is accessible somehow, possibly from the left side menu, eventually removable/hideable by the user. The intro page gives context on the guide itself and how to use it (i.e. that things don’t have to be done in order, things can be hidden, etc.
+
+Below the context is a list of guides split into themes that form a knowledge map. Individual items can be hidden.
+
+Above the guide is a progress bar or other indicator that gives the user some sense of how close to “done” they are with learning the product. This is not an indicator of them being done fully implementing the product.
+
+Within the content we want to weave a concepts in:
+* Tokens
+* Agents
+* Proxy
+* The role of roles (sorry)
+
+Additionally we want to make it possible for the users to self-select relevant content where possible. To add every implementation form for every topic (i.e. for SSH showing scripted setup, IaC setup, point-and-click setup) would be difficult to maintain and confusing. The point is to be opinionated. By showing 1-2 of our implementation details to each guide users, even if just by reading some, can get a sense of the whole and start to figure out how they want to do things for real. To help users self-select and navigate to relevant content, we’ll tag each guide with methods covered. Within the content we can make nods to other methods (i.e. “you could use these installation steps in a cloudinit script in IaC”).
+
+Tags:
+* UI
+  * Connect
+  * CLI
+  * Console
+* Technology
+  * Linux scripting
+  * Infrastructure-as-Code
+  * Kubernetes CRDs
+  * Point-and-Click
+* Team relevance
+  * Security
+  * Compliance
+  * FIPS/FedRAMP
+  * Scalability
+
+### Rough outline of the content
+* Quickstart
+  * You are a local user, what does that mean?
+  * You have the “access” role, what does that mean?
```

Comment:
We should also explain the concept of built-in/Teleport-managed roles vs custom roles that you build yourself.

+1 if we can find a way to mark built-in roles somehow as being "different" within the product and encouraging their use over custom roles wherever appropriate.

Reply:
Yeah we should definitely explain the difference. What roles do you think should be recommended over custom ones? My expectation is we should be recommending custom ones. The built-in ones tend to be really dangerously broad to me, but interested in your thoughts here.